lib/grape/router/pattern.rb

```diff
@@ -0,0 +1,55 @@
+require 'forwardable'
+require 'mustermann/rails'
+
+module Grape
+  class Router
+    class Pattern
+      DEFAULT_PATTERN_OPTIONS   = { uri_decode: true }.freeze
+      DEFAULT_SUPPORTED_CAPTURE = [:format, :version].freeze
+
+      attr_reader :origin, :path, :capture, :pattern
+
+      extend Forwardable
+      def_delegators :pattern, :named_captures, :params
+      def_delegators :@regexp, :===
+      alias_method :match?, :===
+
+      def initialize(pattern, options = {})
+        @origin  = pattern
+        @path    = build_path(pattern, options)
+        @capture = extract_capture(options)
+        @pattern = Mustermann::Rails.new(@path, pattern_options)
```

Comment:
Actually I think we explicitly *don't* want Rails semantics to be inherited because of https://github.com/rkh/mustermann/blob/master/mustermann-rails/lib/mustermann/rails.rb#L38, which is very Rails-version specific. I think you can safely commit a copy of that rails.rb minus everything we don't need into Grape.